python/tests/unit/arcticdb/version_store/test_query_builder.py

```diff
@@ -116,6 +116,107 @@ def test_querybuilder_date_range_then_groupby(lmdb_version_store_tiny_segment, u
     assert_frame_equal(expected, received)
 
 
+@pytest.mark.parametrize("use_row_range_clause", [True, False])
+def test_querybuilder_row_range(lmdb_version_store_tiny_segment, use_row_range_clause):
+    lib = lmdb_version_store_tiny_segment
+    symbol = "test_querybuilder_row_range_then_filter"
+    df = pd.DataFrame({"col1": np.arange(10), "col2": np.arange(100, 110)}, index=np.arange(10))
+    lib.write(symbol, df)
+
+    row_range = (3, 7)
+
+    q = QueryBuilder()
+    if use_row_range_clause:
+        q = q._row_range(row_range)
+
+    if use_row_range_clause:
+        received = lib.read(symbol, query_builder=q).data
+    else:
+        received = lib.read(symbol, row_range=row_range, query_builder=q).data
```

Comment:
Good to know that an empty `QueryBuilder` doesn't blow up. Which code path does it take in this case? `read_direct` or `read_and_process`?

Reply:
Either way, I think the test with `use_row_range_clause==False` should not pass in a query builder, as this is how users would generally approach things.